### PR TITLE
imp PostOpAssign statement

### PR DESCRIPTION
**Pull request by morphey-tech**

### CODE REVIEW

File: tests/all.cs [hunk 1]
```diff
@@ -876,6 +876,191 @@ public void TestAdd()
...
+      string k
+      k += ""foo""
+    }
+    ";
+
+    AssertError<UserError>(
+      delegate() { 
+        Interpret(bhl);
+      },
+      " : PostOpAssign not supported for string"
```
Comment (pachanga) [inline tests/all.cs:979]: let's change the error message: Invalid operand types

File: tests/all.cs [hunk 2]
```diff
@@ -876,6 +876,191 @@ public void TestAdd()
...
+    );
+  }
+
+  [IsTested()]
+  public void TestPostOpAssignExpIncompatibleTypesNotAllowed()
```
Comment (pachanga) [inline tests/all.cs:1044]: We also need a test for the following case:

float k = 2.1
int a = 1
k += a

In this case types are compatible

**Review by pachanga — CHANGES_REQUESTED** [review]

File: src/frontend.cs [hunk 3]
```diff
@@ -1137,7 +1137,7 @@ public override object VisitVarPostOpAssign(bhlParser.VarPostOpAssignContext ctx
       FireError(Location(ctx.NAME()) + " : symbol not resolved");
 
     if(SymbolTable.IsRelopCompatibleType(vlhs.type.type) == false)
-      throw new UserError(Location(ctx.NAME()) + " : PostOpAssign not supported for string");
+      throw new UserError(Location(ctx.NAME()) + " : Invalid operand types");
```
Comment (pachanga) [inline src/frontend.cs:1140]: 2 notes: 
 1) error messages start with lowercase 
 2) let's make it even more specific and clear

So it rather should be: incompatible variable type

File: src/frontend.cs [hunk 4]
```diff
@@ -1137,7 +1137,7 @@ public override object VisitVarPostOpAssign(bhlParser.VarPostOpAssignContext ctx
       FireError(Location(ctx.NAME()) + " : symbol not resolved");
 
     if(SymbolTable.IsRelopCompatibleType(vlhs.type.type) == false)
```
Comment (pachanga) [inline src/frontend.cs:1139]: Why not:

if(!SymbolTable.IsRelopCompatibleType(vlhs.type.type))

?

Comment (morphey-tech) [reply src/frontend.cs:1139]: symbol "!" difficult to notice among the rest,
I considered an explicit comparison cleaner

Comment (pachanga) [reply src/frontend.cs:1139]: Well, this style is not in coding conventions for this project, sorry. I'd like to keep the codebase consistent.

**Review by pachanga — CHANGES_REQUESTED** [review]

File: src/frontend.cs (hunk 4 above)
Comment (morphey-tech) [reply src/frontend.cs]: ok

**Pull request merged**